scipy/ndimage/tests/test_ndimage.py

```diff
@@ -4390,6 +4390,26 @@ def test_white_tophat02(self):
                                       structure=structure)
         assert_array_almost_equal(expected, output)
 
+    def test_white_tophat03(self):
```

Comment:
Could we add a similar test where `input` is a bool array, but `output = np.empty_like(input, dtype=float`? It will detect the potential failure I commented about in the implementation of this function.